### PR TITLE
feature: mark ZAP as lazy

### PR DESCRIPTION
**Pull request by hsiliev**
Do not load ZAP when fetching SD template mappings. Reduces SQL queries twice (from 11 to  5)

Solves: PZ-4189

### CODE REVIEW

File: src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt [hunk 1]
```diff
@@ -119,7 +119,7 @@ class SmartDocumentsTemplatesService @Inject constructor(
                                 builder.and(
                                     builder.equal(
                                         root.get<ZaakafhandelParameters>(
-                                            SmartDocumentsTemplate::zaakafhandelParameters.name
+                                            SmartDocumentsTemplateGroup::zaakafhandelParameters.name
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt:122]: Why was this change necessary? And/or: why did the functionality even work before this change?

**Review by edgarvonk — APPROVED** [review]
Looks good. Just one question.

File: src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt (hunk 1 above)
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt:122]: Because the resulting string is still "zaakafhandleParameters" and cause both object have the field and column with the same name.

**Review by bas-info-nl — APPROVED** [review]
Nice, and no issues with update queries? These lazy loaded objects always cause issues when the object is not within the scope of a Hibernate transaction anymore...

**Comment by hsiliev** [comment]
> Nice, and no issues with update queries? These lazy loaded objects always cause issues when the object is not within the scope of a Hibernate transaction anymore...

Yes indeed. That's why I haven't merged this one yet.

Tested creating new ZAP and updating existing one - no issues so far. However want to do some more exploratory testing Monday to be sure.
